src/xbot/common/math/PID.java

```diff
@@ -23,6 +28,37 @@ public void reset()
     {
         m_prevError = 0;
         m_totalError = 0;
+        errorIsSmall = false;
+        derivativeIsSmall = false;
+    }
+    
+    /**
+     * Set how close the error can be before it is considered
+     * "on-target."
+     *            
+     * This is in the same units as your current and goal values.
+     */
+    public void setErrorTolerance(double errorTolerance) {
```

Comment:
@WasabiFan  Since the error tolerance are linked to DoubleProperties, we're constrained by what we can interact with on the SmartDashboard.

I wanted the ability to turn constraint checking on and off via the SmartDashboard, and to control the constraints themselves. Combining them so that negative values cause constraint checking to be skipped helps reduce clutter a bit.

On the other hand, we currently have two constraints, and may add a third (time). Adding three more enable/disable properties would seem kind of messy to display on the SD. Is there a better solution?